Tests/RequestInterceptorTests.swift

```diff
@@ -117,6 +117,36 @@ final class AdapterTestCase: BaseTestCase {
         // Then
         XCTAssertEqual(result, .doNotRetry)
     }
+
+    func testThatAdapterCanBeImplementedAsynchronously() {
+        // Given
+        let urlRequest = URLRequest(url: URL(string: "https://httpbin.org/get")!)
+        let session = Session()
+        var adapted = false
+
+        let adapter = Adapter { request, _, completion in
+            adapted = true
+            DispatchQueue.main.async {
+                completion(.success(request))
+            }
+        }
+
+        var result: Result<URLRequest>!
+
+        let completesExpectation = expectation(description: "adapter completes")
+
+        // When
+        adapter.adapt(urlRequest, for: session) {
+            result = $0
+            completesExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 0.01)
```

Comment:
Please use our default timeout. It should be available as `timeout`.

Reply:
Oh, sure thing. 🙂